sdv/multi_table/utils.py

```diff
@@ -97,31 +89,50 @@ def _simplify_relationships(metadata, root_table, descendants_to_keep):
         is_parent_in_descendants = parent_table in descendants_to_keep
         is_valid_parent = is_parent_root or is_parent_in_descendants
         is_valid_child = child_table in descendants_to_keep
-        if not is_valid_parent or not is_valid_child:
-            metadata.remove_relationship(parent_table, child_table)
-        else:
+        if is_valid_parent and is_valid_child:
             if is_parent_root:
                 children.append(child_table)
             else:
                 grandchildren.append(child_table)
```

Comment:
I think I misunderstood what this was doing before. It's fine to prune the metadata here while you're already looping through the relationships. Can we update the docstring though to reflect all of that?